mlrun/utils/notifications/notification/mail.py

```diff
@@ -0,0 +1,141 @@
+                    raise ValueError(
+                        "Parameter 'email_addresses' must be a string or a list of strings"
+                    )
+
+                if not re.match(mlrun.utils.regex.mail_regex, email_address):
+                    raise ValueError(
+                        f"Invalid email address '{email_address}' in 'email_addresses'"
+                    )
+
+    async def push(
+        self,
+        message: str,
+        severity: typing.Optional[
+            typing.Union[mlrun.common.schemas.NotificationSeverity, str]
+        ] = mlrun.common.schemas.NotificationSeverity.INFO,
+        runs: typing.Optional[typing.Union[mlrun.lists.RunList, list]] = None,
+        custom_html: typing.Optional[typing.Optional[str]] = None,
+        alert: typing.Optional[mlrun.common.schemas.AlertConfig] = None,
+        event_data: typing.Optional[mlrun.common.schemas.Event] = None,
+    ):
+        self.params.setdefault("subject", f"[{severity}] {message}")
+        self.params.setdefault("body", message)
+        await self._send_email(**self.params)
+
+    @staticmethod
+    async def _send_email(
+        email_addresses: str,
+        sender_address: str,
+        server_host: str,
+        server_port: int,
+        username: str,
+        password: str,
+        use_tls: bool,
+        start_tls: bool,
+        validate_certs: bool,
+        subject: str,
+        body: str,
+        **kwargs,
+    ):
+        # Create the email message
+        message = EmailMessage()
+        message["From"] = sender_address
+        message["To"] = email_addresses
+        message["Subject"] = subject
+        message.set_content(body)
+
+        # Send the email
+        await aiosmtplib.send(
+            message,
+            hostname=server_host,
+            port=server_port,
+            username=username,
+            password=password,
+            use_tls=use_tls,
+            validate_certs=validate_certs,
+            start_tls=start_tls,
+        )
+
+    @classmethod
+    def enrich_default_params(
+        cls, params: dict, default_params: typing.Optional[dict] = None
+    ) -> dict:
+        params = super().enrich_default_params(params, default_params)
+
+        cls._set_param_default_value(params, "use_tls", "true")
+        cls._set_param_default_value(params, "start_tls", "false")
+        cls._set_param_default_value(params, "validate_certs", "true")
+
+        params.setdefault("server_port", DEFAULT_SMTP_PORT)
+
+        default_mail_address = params.pop("default_email_addresses", "")
+        email_addresses = params.get("email_addresses", default_mail_address)
+        if type(email_addresses) is list:
+            email_addresses = ",".join(email_addresses)
+        params["email_addresses"] = email_addresses
+
+        return params
+
+    @staticmethod
+    def _set_param_default_value(params, key, default_value):
+        if type(params.get(key, "")) is str:
```

Comment:
I think it's slightly mot pythonic to do 
`isinstance(params.get(key, ""),str)`